client/dev-tools/components/dev-tools.tsx

```diff
@@ -33,8 +33,10 @@ const PromoCard = ( { title, text, supportContext }: PromoCardProps ) => (
 );
 
 const DevTools = () => {
-	const [ showEligibility, setShowEligibility ] = useState( false );
+	const { searchParams } = new URL( document.location.toString() );
 	const siteId = useSelector( getSelectedSiteId );
+	const redirectUrl = useRef( searchParams.get( 'redirect_to' ) ?? `/hosting-config/${ siteId }` );
```

Comment:
Why is `useRef` needed here?

Reply:
I think I know why... it is because `redirect_to` is not included here, so the param will be gone as soon as it is mounted.

https://github.com/Automattic/wp-calypso/blob/c640cff1674b46c767f19a31e54358d1822507ad/client/sites-dashboard-v2/controller.tsx#L19-L26

A simpler solution is maybe to add that query param in the above logic 🤔

Reply:
Another solution is get the URL param out of the functional component to avoid losing it after renders.

Why not just use the ref?

Reply:
> Why not just use the ref?

I don't have strong feeling against it. Just it could confuse future developers as it's not that obvious why it's needed here. If we want to use it, I suggest we add some code comment 🤔

Reply:
I tried adding the param `redirect_to` to `getQueryParams` but it didn't work, so I'm adding a comment to keep the ref.